Funcionalidades do diário e conteúdo

eumsmo opened the pull request:
# [Grupo 4 - E] Funcionalidades do diário e conteúdo

* Todas funcionalidades da tabela diário foram implementadas
* Todas funcionalidades da tabela conteúdo foram implementadas
* Arquivos modificados para seguir o padrão geral
* Ao deletar um conteúdo/atividade, o diário relativo a ele também é deletado

## Observações

A unica coisa que falta, ao meu ver, é checar se a funcionalidade pode ser feita, por exemplo, se alguém lança notas em um conteúdo específico, checar se o conteúdo realmente existe. Outro exemplo seria checar se o usuário que está logado pode efetuar tal operação, e se sim quais suas limitações.

eumsmo committed:
📚 Desprivatização do Descartes

- Adicionado Consulta e Inserir do Descartes
- Adicionado classe RespostaXML para gerar XML a partir do formato especificado

NOTA: Recursos que precisavam especificamente do repositório "servidor" ainda não foram implementados por favor aceita meu commit

eumsmo committed:
🔧 Ajuste na conexão ao servidor

- Linhas modificadas para ajustar a conexão ao servidor proposta pela gerência

NOTA: não consegui testar mas me parece certo

eumsmo committed:
📒 Desprivatização do diário

- Primeiro commit da parte de diário
- Inserir e consultar conteudos
- Inserir atividades

NOTA: está faltando a Falta pois a tabela não faz sentido e estamos vendo com a gerência sobre isso

eumsmo committed:
Merge branch 'dev-s1' into feature/diario/diario

eumsmo committed:
Implementação de Insere e Consulta Conteudo/Atividade

E inicio do atualizar

eumsmo committed:
📒 Conteúdo e Atividade parcialmente concluído

- Levando em conta que conteúdo e atividade se tornaram a mesma tabela...
- CRUD de Conteúdo e Atividade criados

NOTA: Por mais que pareça estar 100%, ainda não toquei na parte de autenticação do usuário... E ainda falta as faltas.

eumsmo committed:
🧐 Dedo mindinho levantado

É a mesma coisa que o ultimo commit mas com um toque mais refinado.

- Foi adicionado mais filtros de parâmetros nas requisições
- Falha de conexão no banco de dados mais evidente
- Consulta finalmente faz algum sentido

NOTA: Ainda falta a autenticação do usuário!

eumsmo committed:
Merge branch 'dev-s1' into feature/diario/diario

eumsmo committed:
👨‍🎓 Começo do lançando as decepções

- Adicionado o lançamento de notas e faltas (ainda não finalizado)
- Pequenas correções no sistema de lançar atividades e conteúdos

NOTA: Pode parecer que deixar o sistema de autenticação pro final vai prejudicar tanto, mas confia que não vai. Espero que em um commit futuro ambos sistemas estarão mais refinados, checando a existência dos ids mandado pelo usuário e respondendo diferente dependendo de qual usuário está conectado

eumsmo committed:
👨‍🏫 Lançando tudo que você imaginar 👨‍🎓

- Organização nos arquivos
- Funcionalidades do diário 100% funcional
- Funcionalidades do conteúdo 100% funcional

NOTA: Falta apenas e exclusivamente apenas o gerenciamento de usuário e algumas pequenas verificações.

LiviaLelis commented on app/nbproject/project.properties:
```diff
@@ -1,87 +1,94 @@
+war.ear.name=${war.name}
+war.name=app.war
+web.docbase.dir=web
+webinf.dir=web/WEB-INF/lib
```
```suggestion
webinf.dir=web/WEB-INF
```

LiviaLelis commented on app/nbproject/project.properties:
```diff
@@ -1,87 +1,94 @@
+annotation.processing.processors.list=
+annotation.processing.run.all.processors=true
+annotation.processing.source.output=${build.generated.sources.dir}/ap-source-output
+auxiliary.org-netbeans-modules-css-prep.less_2e_compiler_2e_options=
```
Desfazer as mudanças das linhas 6 a 12

LiviaLelis commented on app/src/java/biblioteca/descartes/Consulta.java:
```diff
@@ -0,0 +1,105 @@
+     * @throws IOException if an I/O error occurs
+     */
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
```
_Sugestão:_ Não precisa cumprir `POST`

LiviaLelis commented on app/src/java/biblioteca/descartes/Inserir.java:
```diff
@@ -0,0 +1,110 @@
+	    try {
+		// Query SQL de inserção na tabela DESCARTES
+		String query = "INSERT INTO descartes(`id-acervo`,`data-descarte`, motivos, operador) VALUES (?,?,?,?)";
+		String queryDelete = "DELETE FROM acervo WHERE id=" + id_acervo;
```
Não sei exatamente a regra de negócio, mas ok...

LiviaLelis commented on app/src/java/biblioteca/descartes/Inserir.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws IOException if an I/O error occurs
+     */
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
```
_Sugestão:_ Não precisa cumprir `GET`

LiviaLelis commented on app/src/java/diario/diario/conteudos/ChecaParametro.java:
```diff
@@ -0,0 +1,50 @@
+        return false;
+    }
+    
+    public static boolean parametroNaoVazio(HttpServletRequest req, String parametro){
```
Não era pra haver algum teste aqui?

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+    /* Inserir */
+
+    public static void insere(int idEtapas, int idDisciplinas, String conteudo, Date data ,Double nota) throws ErroException, SQLException{
+        update("INSERT INTO conteudos(`id-etapas`,`id-disciplinas`, conteudos , data, valor) VALUES ("+idEtapas+","+idDisciplinas+",\""+conteudo+"\",\""+data.toString()+"\","+nota+")");
```
Query vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+    }
+    
+    public static void insere(int idEtapas, int idDisciplinas, String conteudo, Date data) throws ErroException, SQLException{
+        update("INSERT INTO conteudos(`id-etapas`,`id-disciplinas`, conteudos , data, valor) VALUES ("+idEtapas+","+idDisciplinas+",\""+conteudo+"\",\""+data.toString()+"\", 0.0)");
```
Query vulnerável e o parâmetro valor é opcional, não precisando ser explicitado na última versão das tabelas

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+        if(modificacoes.size()==0)
+            throw new ErroException("Nenhuma alteração detectada!","Nenhum parametro de modificação recebido");
+        
+        update("UPDATE conteudos SET "+String.join(",",modificacoes)+" WHERE id="+id);
```
Query vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+            DiarioRepository.removerPorConteudo(id);
+        }
+        
+        update("DELETE FROM conteudos WHERE `id-etapas`="+idEtapa);
```
Query vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+            DiarioRepository.removerPorConteudo(id);
+        }
+        
+        update("DELETE FROM conteudos WHERE `id-disciplinas`="+idDisciplina);
```
Query vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+            DiarioRepository.removerPorConteudo(id);
+        }
+        
+        update("DELETE FROM conteudos WHERE `id-etapas`="+idEtapa +" AND `id-disciplinas`="+idDisciplina);
```
Query vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+        if(filtro.size()==0)
+            throw new ErroException("Nenhum parametro detectado!","Nenhum parametro de consulta recebido");
+        
+        return execute("SELECT * FROM conteudos WHERE "+String.join(" AND ",filtro)+complemento);
```
Query pode vir a ser vulnerável

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+        if(conexao==null)
+            throw new ErroException("Falha na conexão!","Falha em tentar conectar com o banco de dados");
+        
+        PreparedStatement st = conexao.prepareStatement(query);
```
??????????????????????????????????

LiviaLelis commented on app/src/java/diario/diario/conteudos/ConteudosRepository.java:
```diff
@@ -0,0 +1,203 @@
+public class ConteudosRepository {
+    /* Utils */
+
+    public static void update(String query) throws ErroException, SQLException{
```
Hmmmmmmmmmmmmmmm

LiviaLelis commented on app/src/java/diario/diario/diario/Atualizar.java:
```diff
@@ -0,0 +1,119 @@
+import diario.diario.utils.ErroException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import static java.lang.System.out;
```
Isso não é um bom sinal, se confunde com o `PrintWriter out`

LiviaLelis commented on app/src/java/diario/diario/diario/Consulta.java:
```diff
@@ -0,0 +1,123 @@
+                out.print(DiarioView.consultaSemFalta(resultado));
+                
+            } else
+                throw new ErroException("Acesso negado!","O acesso a toda tabela é um recurso exclusivo do Administrador");
```
???????????????????????????????????????????

LiviaLelis commented on app/src/java/diario/diario/utils/ErroException.java:
```diff
@@ -0,0 +1,31 @@
+ *
+ * @author juanr
+ */
+public class ErroException extends Exception{
```
Hmmmmmmmmmmmmmmm

LiviaLelis reviewed (changes requested):
## Correções gerais

- Diversos problemas de formatação, reformatar usando as ferramentas da IDE usada, verificando se está sendo utilizado `TABS`
- Autenticação ausente em rotas que devem ser protegidas
- _Não é necessário_, mas o nome das classes dos Servlets costumam ser mais específicos dentro do projeto
- Não apontarei a maioria das `code smells`, por mais que presentes
- Quebrei um pouquinho o item acima
- Fui cansando de revisar a medida que os erros se repetiam, mas o que se refere a uma parte deve se referir à outra também

eumsmo committed:
⏰ É questão de tempo

Comecei a colocar os arquivos no formato adequado, não tão todos e eu sei que tem erro, eu SEI disso, é que eu vou terminar depois

Commit:
Revert build.xml

Commit:
Revert ant-deploy.xml

Commit:
Revert project.properties

Commit:
Revert project.xml

eumsmo committed:
👨‍🏫 Diário no formato

Código do Diário atualizado a um formato aceitável pela sociedade opressora.

NOTA: Falta colocar Conteudos no formato e implementar autenticação.

eumsmo committed:
Merge branch 'dev-s1' into feature/diario/diario

eumsmo committed:
👨‍🏫 Lanço notas e faltas no estilo padrãozinho👨‍🎓

- Me torno um projeto padrãozinho com cabelo degradê, espero que a sociedade me aceite
- Inicio de autenticação

NOTA: queria deixar BEM CLARO que não é só isso de autenticação que vai ter, é só que não tem como fazer todo o projeto em um commit só, sem tempo

Commit:
Merge branch 'dev-s1' of github.com:INF2-2019/servidor into feature/diario/diario

Commit:
fix formatting
